src/outputs/history.cpp

```diff
@@ -49,68 +49,113 @@ void HistoryOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm,
   if (signal == SignalHandler::OutputSignal::now) {
     return;
   }
-  std::vector<std::string> all_labels = {};
-  std::vector<Real> all_results = {};
+
+  std::map<UserHistoryOperation, std::vector<Real>> results;
+  std::map<UserHistoryOperation, std::vector<std::string>> labels;
+  std::vector<UserHistoryOperation> ops = {
+      UserHistoryOperation::sum, UserHistoryOperation::max, UserHistoryOperation::min};
+  for (auto &op : ops) {
+    results[op] = {};
+    labels[op] = {};
+  }
+
+  // If "packages" not provided, output history for all packages
+  auto &requested_packages = output_params.packages;
+  Dictionary<std::shared_ptr<StateDescriptor>> packages;
+  if (requested_packages.empty()) {
+    packages = pm->packages.AllPackages();
+  } else {
+    const auto &all_packages = pm->packages.AllPackages();
+    for (const auto &pkg_name : requested_packages) {
+      const auto &it = all_packages.find(pkg_name);
+      if (it != all_packages.end()) {
+        packages[(*it).first] = (*it).second;
```

Comment:
Should we throw an error if the packages has not been found? To prevent accidental typos in the input file.

Reply:
Yeah I agree. One could imagine someone wanting to write out history from specific packages and automatically not getting that output if they enable/disable individual packages at runtime, but I think catching typos is more important. Also, a way to disable packages at runtime is to enroll an empty package with the same name, which would work for the above. 

Changed to error if this `if` statement is `false`.